### DISCUSSION TITLE
Failing to load image

### DISCUSSION THREAD

**Discussion by rousseauolivier** [mpdf/mpdf#1643] [Q&A]
Hello,

I have a mpdf script that was working wonderfully, for months on my app until this week.

It generates the PDF as expected from an HTML file, but it fails to load a picture that I fetch from AWS S3.
It just appears the blue question mark icon instead of the image.

-- Additional info --
I got other images on the PDF but the only one failing to load correctly is one that is being fetched from AWS S3.
The other ones are loading correctly and are being stored on the same server as the MPDF script.

I copy-pasted my code locally and the image is being loaded correctly.
-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --

I have not changed any files related to the script or whatsoever and I am very confused why this is happening now.

**Comment by finwe** [comment]
Seems like an issue on the remote side. Have you tried all debugging options available? `Mpdf::$debug`, `Mpdf::$showImageErrors`, [Logging](https://mpdf.github.io/installation-setup/logging.html)?

**Reply by rousseauolivier** [reply]
I've put logging options, but the debug file is empty

**Comment by jakejackson1** [comment]
[This is due to security enhancements in mPDF](https://github.com/mpdf/mpdf/issues/991). You need to add the `s3` file wrapper to the allowed stream wrappers: https://github.com/mpdf/mpdf.github.io/commit/efea96a86cfd128e6ffd761d66406eab40df6623

**Reply by rousseauolivier** [reply]
I don't think this would be the issue because I use the public file URL (all image assets are public) and it just started happening last week or so

**Reply by jakejackson1** [reply]
Okay. You'll need to debug more as there isn't enough information given to provide any further advice.

**Reply by rousseauolivier** [reply]
I could send you the html content, but I don't think this would help since it works locally.
What different configuration could make this happen on a distant server?

**Reply by Klap-in** [reply]
Which mpdf version are you using? Recent versions have also changes in the remote loading. Could you please find a minimal case?
